src/coreclr/vm/amd64/JitHelpers_InlineGetThread.asm

```diff
@@ -115,6 +118,7 @@ align 16
 
     ClassNotInited:
     AllocFailed:
+    NullRef:
```

Comment:
You can delete `enum_flag_Unrestored` check above while you are on it. It is left-over from fragile NGen support.

Reply:
@davidwrighton and I discussed this a while back. We couldn't confirm this was safe to remove in all cases. The idea being we saw other APIs that made the init check in many places. Can all be removed?

Reply:
The unrestored flag is set only for a brief period of time while the type is being built. Based on the comments, it was a trick to prevent IBC logging during type building. 

Once the type is built, the unrestored flag is always cleared. The type handles that the JIT embeds into the code are always fully restored.